cmd/argo-watcher/server/router.go

```diff
@@ -265,3 +287,75 @@ func (env *Env) healthz(c *gin.Context) {
 func (env *Env) getConfig(c *gin.Context) {
 	c.JSON(http.StatusOK, env.config)
 }
+
+// SetDeployLock godoc
+// @Summary Set deploy lock
+// @Description Set deploy lock
+// @Tags frontend
+// @Success 200 {string} string
+// @Router /api/v1/deploy-lock [post].
+func (env *Env) SetDeployLock(c *gin.Context) {
+	if err := env.validateKeycloakToken(c); err != nil {
+		log.Error().Msgf("couldn't release deploy lock, got the following error: %s", err)
+		c.JSON(http.StatusUnauthorized, models.TaskStatus{
+			Status: unathorizedMessage,
+		})
+		return
+	}
+
+	env.deployLockSet = true
+
+	log.Debug().Msg("deploy lock is set")
+	c.JSON(http.StatusOK, "deploy lock is set")
+}
+
+// ReleaseDeployLock godoc
+// @Summary Release deploy lock
+// @Description Release deploy lock
+// @Tags frontend
+// @Success 200 {string} string
+// @Router /api/v1/deploy-lock [delete].
+func (env *Env) ReleaseDeployLock(c *gin.Context) {
+	if err := env.validateKeycloakToken(c); err != nil {
+		log.Error().Msgf("couldn't release deploy lock, got the following error: %s", err)
+		c.JSON(http.StatusUnauthorized, models.TaskStatus{
+			Status: unathorizedMessage,
+		})
+		return
+	}
+
+	env.deployLockSet = false
+
+	log.Debug().Msg("deploy lock is released")
+	c.JSON(http.StatusOK, "deploy lock is released")
+}
+
+// isDeployLockSet godoc
+// @Summary Check if deploy lock is set
+// @Description Check if deploy lock is set
+// @Tags frontend
+// @Success 200 {boolean} boolean
+// @Router /api/v1/deploy-lock [get].
+func (env *Env) isDeployLockSet(c *gin.Context) {
+	c.JSON(http.StatusOK, env.deployLockSet)
+}
+
+func (env *Env) validateKeycloakToken(c *gin.Context) error {
+	keycloakToken := c.GetHeader("Authorization")
+
+	if keycloakToken != "" {
```

Comment:
Should we check if Keycloak is enabled there as well?

Reply:
Not sure, probably it would be better not to trigger this method at all if Keycloak integration is not enabled. But Keycloak is not a part of the changes from this PR. Let's move this discussion to issues.